### PR TITLE
Issue 8873: Add XOR and Hash tests for Database Store

### LINKED ISSUE #8873: Add a unit test for client secret hashing for DatabaseStore
The existing unit test for the databaseStore client, `CachedDBOidcClientProviderTest ` does not store the metadata.

In order to make a hashed client secret, we'll have to add more infrastructure to the test in order to store the metatype (so the hash information can be stored as well).

Keep the XOR/original test. To match the CustomStore OauthClientStoreTest, rename `CachedDBOidcClientProviderTest` to `CachedDBOidcClientProviderXORTest` and then create a `CachedDBOidcClientProviderHashTest`

### PR DESCRIPTION
**Pull request by kristip17**
Fixes #8873 

Added a common unit test for Database store
Added an XOR and Hash version (previously only the XOR version)
Updated the Mock database to store the clientmetadata

### CODE REVIEW

**Comment by kristip17** [comment]
#build

**Comment by kristip17** [comment]
Running static build here -- https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_2oDToNDYEemo7shcvMe-cQ

**Comment by kristip17** [comment]
Confirmed that XOR and Hash test for Database store ran on the PB and the static build.

File: dev/com.ibm.ws.security.oauth/test/com/ibm/ws/security/oauth20/plugins/db/CachedDBOidcClientProviderXORTest.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        outputMgr.captureStreams();
+
+        System.setProperty(Context.INITIAL_CONTEXT_FACTORY, InitialContextFactoryMock.class.getName());
+
+        SAMPLE_CLIENTS = getsampleOidcBaseClients(5, PROVIDER_NAME);
+    }
+
+    @Override
+    @Before
+    public void setupBefore() {
```
Comment (ayoho) [inline dev/com.ibm.ws.security.oauth/test/com/ibm/ws/security/oauth20/plugins/db/CachedDBOidcClientProviderXORTest.java]: It looks like this method could actually stay in the abstract `CachedDBOidcClientProviderTest` as-is; the appropriate overridden `invokeConstructorAndInitialize()` method would still be invoked at runtime, so everything should execute correctly.

**Comment by kristip17** [comment]
Review comments addressed in #9175

**Pull request merged**